fhir-model/src/main/java/com/ibm/fhir/model/util/ValidationSupport.java

```diff
@@ -106,17 +123,29 @@ public static void checkString(String s) {
         int count = 0;
         for (int i = 0; i < s.length(); i++) {
             char ch = s.charAt(i);
+            Character character = Character.valueOf(ch);
+            checkUnsupportedUnicode(s, ch);
             if (!Character.isWhitespace(ch)) {
                 count++;
-            } else if (!WHITESPACE.contains(ch)) {
-                throw new IllegalStateException(String.format("String value: '%s' is not valid with respect to pattern: [ \\r\\n\\t\\S]+", s));
+            } else if (!WHITESPACE.contains(character)) {
+                throw new IllegalStateException(String.format("String value: '%s' is not valid with respect to pattern: [\\r\\n\\t\\S]+", s));
             }
         }
         if (count < MIN_STRING_LENGTH) {
             throw new IllegalStateException(String.format("Trimmed String value length: %d is less than minimum required length: %d", count, MIN_STRING_LENGTH));
         }
     }
 
+    /**
+     * Helper method to check if there is unsupported unicode.
+     * @param ch
+     */
+    private static void checkUnsupportedUnicode(String s, Character ch) {
+        if (UNSUPPORTED_UNICODE.contains(ch)) {
```

Comment:
The key check

Reply:
Is there a micro-benchmark for this? I'm concerned about performing a Set lookup for every inbound character. I think John had some benchmarking stuff for the model parsing, perhaps that would be a good reference.

Reply:
FHIRParserBenchmark (under `fhir-benchmark`) could probably be used for this.  I think it defaults to using different resources each time, but there's options to feed it a specific set of resources (either in main or maybe I just have that locally).

Reply:
I'm currently running the FHIRParserBenchmark. 
I'll report back and change if needed.